packages/core/src/MatchMaker.ts

```diff
@@ -245,17 +245,17 @@ export function defineRoomType<T extends Type<Room>>(
   handlers[name] = registeredHandler;
 
   if (!isDevMode) {
-    cleanupStaleRooms(name);
+    await cleanupStaleRooms(name);
   }
 
   return registeredHandler;
 }
 
-export function removeRoomType(name: string) {
+export async function removeRoomType(name: string) {
   delete handlers[name];
 
   if (!isDevMode) {
-    cleanupStaleRooms(name);
+    await cleanupStaleRooms(name);
```

Comment:
seems unnecessary. Cleaning up stale rooms should be handled asynchronously.
I think using `.catch` might be useful, but why wait?